static/app/components/smartSearchBar/index.tsx

```diff
@@ -267,14 +271,19 @@ class SmartSearchBar extends React.Component<Props, State> {
     this.inputResizeObserver.observe(this.containerRef.current);
   }
 
-  componentDidUpdate(prevProps: Props) {
+  componentDidUpdate(prevProps: Props, prevState: State) {
     const {query} = this.props;
     const {query: lastQuery} = prevProps;
 
     if (query !== lastQuery && defined(query)) {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState({query: addSpace(query)});
     }
+
+    if (this.state.query !== prevState.query) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({parsedQuery: parseSearch(this.state.query)});
```

Comment:
prob unlikely, but will we run into any weird race conditions where the state's query has changed but the state's AST hasn't updated so we end up showing the wrong thing in the autocomplete menu?

Reply:
This should all happen as one set state update I believe.